src/main/java/mate/academy/shop/service/impl/UserServiceImpl.java

```diff
@@ -60,4 +69,26 @@ public User login(String login, String password) throws AuthenticationException
     public Optional<User> getByToken(String token) {
         return userDao.getByToken(token);
     }
+
+    @Override
+    public String hashPassword(String password, byte[] salt) {
```

Comment:
better move this method in the Util class

where do you get salt?

Reply:
I generate it in user constructor, in method login I take it from sql